packages/ddp-client/common/livedata_connection.js

```diff
@@ -850,7 +850,17 @@ export class Connection {
     }
 
     // If we added it to the first block, send it out now.
-    if (self._outstandingMethodBlocks.length === 1) methodInvoker.sendMessage();
+    if (self._outstandingMethodBlocks.length === 1) {
+      if (callback && future) {
+        // Ensure the method message after the result of the method ran in the client.
+        future = new Promise((resolve) => {
```

Comment:
Here you're replacing [this future](https://github.com/meteor/meteor/blob/1a7cb656e2f4da7911ca6ed44c6063302f50078d/packages/ddp-client/common/livedata_connection.js#L803-L813) promise. So when I call an applyAsync, callAsync, or any Mongo db function, for that matter, I won't get the return from the server/stub... this doesn't look correct to me.